firebase-devservices/deployment/src/main/java/io/quarkiverse/googlecloudservices/firebase/deployment/testcontainers/FirebaseEmulatorContainer.java

```diff
@@ -0,0 +1,1382 @@
+            FirebaseConfig firebaseConfig) {
+    }
+
+    // Use node:20 for now because of https://github.com/firebase/firebase-tools/issues/7173
+    /**
+     * The default image to use for building the docker image.
+     */
+    public static final String DEFAULT_IMAGE_NAME = "node:20-alpine";
+    /**
+     * The default version of the firebase tools to install.
+     */
+    public static final String DEFAULT_FIREBASE_VERSION = "latest";
+
+    /**
+     * Builder for the {@link FirebaseEmulatorContainer} configuration.
+     */
+    public static class Builder {
+
+        private DockerConfig dockerConfig = DockerConfig.DEFAULT;
+        private String firebaseVersion = DEFAULT_FIREBASE_VERSION;
+        private CliArgumentsConfig cliArguments = CliArgumentsConfig.DEFAULT;
+
+        private Path customFirebaseJson;
+        private FirebaseConfig firebaseConfig;
+
+        private Builder() {
+        }
+
+        /**
+         * Configure the docker options
+         *
+         * @return THe docker config builder
+         */
+        public DockerConfigBuilder withDockerConfig() {
+            return new DockerConfigBuilder();
+        }
+
+        /**
+         * Configure the firebase version
+         *
+         * @param firebaseVersion The firebase version
+         * @return The builder
+         */
+        public Builder withFirebaseVersion(String firebaseVersion) {
+            this.firebaseVersion = firebaseVersion;
+            return this;
+        }
+
+        /**
+         * Configure the CLI argument options
+         *
+         * @return The CLI Builder
+         */
+        public CliBuilder withCliArguments() {
+            return new CliBuilder();
+        }
+
+        /**
+         * Read the configuration from the custom firebase.json file.
+         *
+         * @param customFirebaseJson The path to the custom firebase json
+         * @return The builder
+         * @throws IOException In case the file could not be read.
+         */
+        public Builder readFromFirebaseJson(Path customFirebaseJson) throws IOException {
+            var reader = new CustomFirebaseConfigReader();
+            this.firebaseConfig = reader.readFromFirebase(customFirebaseJson);
+            this.customFirebaseJson = customFirebaseJson;
+            return this;
+        }
+
+        /**
+         * Configure the firebase emulators
+         *
+         * @return The firebase config builder
+         */
+        public FirebaseConfigBuilder withFirebaseConfig() {
+            return new FirebaseConfigBuilder();
+        }
+
+        /**
+         * Build the configuration
+         *
+         * @return The emulator configuration.
+         */
+        public EmulatorConfig buildConfig() {
+            if (firebaseConfig == null) {
+                // Try to autoload the firebase.json configuration
+                var defaultFirebaseJson = new File("firebase.json").getAbsoluteFile().toPath();
```

Comment:
This will not work in a JAR, you should use `getResourceAsStream()` and reads from META-INF

Reply:
True, but that's actually intentional. In a firebase project (using the Firebase CI tools), a firebase.json file is generated at the root of the project itself (i.e. outside of the JAR). So by using this approach, you actually have the same behavior as the normal Firebase CI tools and the user won't see the difference